.github/workflows/release.yml

```diff
@@ -1,57 +1,104 @@
-name: Release
+name: Release Build and Sign
 
-# Uncomment the following to let goreleaser automatically
-# create a GitHub release when a tag is pushed.
-# permissions:
-#   contents: write
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
 
 on:
-  push:
-    branches-ignore:
-      - '**'
-    tags:
-      - 'v*.*.*'
-      # to be used by fork patch-releases ^^
-      - 'v*.*.*-*'
+  release:
+    types: [created]
   workflow_dispatch:
+  push:
+    branches:
+      - feat/releases
+  #TODO: remove the trigger above!
 
 jobs:
-  goreleaser:
-    runs-on: ubuntu-latest
+  deploy-releaser:
+    runs-on: self-hosted
     steps:
-      - name: Checkout
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Build releaser utility
+        run: |
+          # TODO: remove the full path to go
+          /usr/local/go/bin/go build -o ./releaser ./cmd/release
+
+      - name: Deploy Releaser
+        env:
+          REMOTE_HOST: ${{ secrets.DOWNLOADS_HOST }}
+          REMOTE_USER: ${{ secrets.DEPLOY_USER }}
+          REMOTE_KEY: ${{ secrets.DEPLOY_KEY }}
+          REMOTE_DIR: "~/releaser/"
+        run: |
+          mkdir -p ~/.ssh
+          echo "$REMOTE_KEY" > ~/.ssh/gh_actions_key
+          chmod 600 ~/.ssh/gh_actions_key
+          rsync -avzr --delete -e "ssh -p 22 -i ~/.ssh/gh_actions_key -o StrictHostKeyChecking=no" ./releaser $REMOTE_USER@$REMOTE_HOST:$REMOTE_DIR
+
+  build-and-release:
+    runs-on: self-hosted
+    needs: deploy-releaser
+    strategy:
+      fail-fast: false
+      matrix:
+        binary: [erigon, rpcdaemon]
+        os: [linux] # darwin, windows
```

Comment:
Overall looks good. 

1. To build for various archs - you probably need special compilers/static-libs/etc... who will maintain it? where is infra code? how to change go/clang version? etc... 

2. What is profit of running build on server compare to running inside `docker pull ghcr.io/goreleaser/goreleaser-cross`?

3. Do we need docker images build also? 

4. Do we need `make db-tools` build also?

Reply:
Thanks @AskAlexSharov I didn't realise it as you who added goreleaser, I thought that was someone else! 

1. I think we should use ansible for this and version control alongside the releaser
2. Not sure that goreleaser can get around the cgo restrictions that mean we must build on the specific os/arch - for something as simple as builds I couldn't see the advantage of goreleaser over some home-rolled simple code
3. Docker images I've left in the hands of dockerhub for now - but we could also look to change this
4. Not sure about db-tools - are they requisite for a release?

Reply:
Would be great if we have db-tools in the docker image since we will use the mdbx_copy command more in our daily operations. :pray: